curated_transformers/generation/config.py

```diff
@@ -21,29 +21,34 @@ class GeneratorConfig(ABC):
     """
     Configuration of the generator.
 
+    :param default_logits_transform:
+        The logits transform that are unconditionally applied to all
+        generations. Usually multiple composed transforms, can be
+        empty.
     :param eos_id:
         End-of-sequence identifier that should end the generation of a sequence
         when predicted. When this value is set to `None`, it is the
         responsibility of the generator to set it.
-
     :param max_generated_pieces:
         The maximum number of generation steps. This condition is a noop
         for values less than 1. When this value is set to `None`, it is the
         responsibility of the generator to set it.
     """
 
+    default_logits_transform: LogitsTransform = field(
```

Comment:
This is not how configuration has worked so far. In the current setup, we'd rather add a field like `masked_pieces: Set[int]` and the `logits_transform` method of the particular config would convert this to a `LogitsTransform`.

We should make it possible for the user to directly configure transforms (I have an idea of how we could do this, but this requires changing the class hierarchy, but that's out the scope of this PR). But the main idea behind the current configuration classes is that the user configures concrete options and does not have to bother with the underlying classes (like `LogitsTransform` subclasses).

Reply:
Makes sense. I'll rewrite it as suggested.